langchain/retrievers/kendra.py

```diff
@@ -0,0 +1,280 @@
+            return self.AdditionalAttributes[0].get_value_text()
+
+    def get_excerpt(self) -> str:
+        if (
+            self.AdditionalAttributes
+            and self.AdditionalAttributes[0].Key == "AnswerText"
+        ):
+            excerpt = self.get_attribute_value()
+        elif self.DocumentExcerpt:
+            excerpt = self.DocumentExcerpt.Text
+        else:
+            excerpt = ""
+
+        return clean_excerpt(excerpt)
+
+    def to_doc(self) -> Document:
+        title = self.DocumentTitle.Text
+        source = self.DocumentURI
+        excerpt = self.get_excerpt()
+        type = self.Type
+        page_content = combined_text(title, excerpt)
+        metadata = {"source": source, "title": title, "excerpt": excerpt, "type": type}
+        return Document(page_content=page_content, metadata=metadata)
+
+
+class QueryResult(BaseModel, extra=Extra.allow):
+    ResultItems: List[QueryResultItem]
+
+    def get_top_k_docs(self, top_n: int) -> List[Document]:
+        items_len = len(self.ResultItems)
+        count = items_len if items_len < top_n else top_n
+        docs = [self.ResultItems[i].to_doc() for i in range(0, count)]
+
+        return docs
+
+
+class DocumentAttributeValue(BaseModel, extra=Extra.allow):
+    DateValue: Optional[str]
+    LongValue: Optional[int]
+    StringListValue: Optional[List[str]]
+    StringValue: Optional[str]
+
+
+class DocumentAttribute(BaseModel, extra=Extra.allow):
+    Key: str
+    Value: DocumentAttributeValue
+
+
+class RetrieveResultItem(BaseModel, extra=Extra.allow):
+    Content: Optional[str]
+    DocumentAttributes: Optional[List[DocumentAttribute]] = []
+    DocumentId: Optional[str]
+    DocumentTitle: Optional[str]
+    DocumentURI: Optional[str]
+    Id: Optional[str]
+
+    def get_excerpt(self) -> str:
+        if not self.Content:
+            return ""
+        return clean_excerpt(self.Content)
+
+    def to_doc(self) -> Document:
+        title = self.DocumentTitle if self.DocumentTitle else ""
+        source = self.DocumentURI
+        excerpt = self.get_excerpt()
+        page_content = combined_text(title, excerpt)
+        metadata = {"source": source, "title": title, "excerpt": excerpt}
+        return Document(page_content=page_content, metadata=metadata)
+
+
+class RetrieveResult(BaseModel, extra=Extra.allow):
+    QueryId: str
+    ResultItems: List[RetrieveResultItem]
+
+    def get_top_k_docs(self, top_n: int) -> List[Document]:
+        items_len = len(self.ResultItems)
+        count = items_len if items_len < top_n else top_n
+        docs = [self.ResultItems[i].to_doc() for i in range(0, count)]
+
+        return docs
+
+
+class AmazonKendraRetriever(BaseRetriever):
+    """Retriever to query documents from Amazon Kendra Index.
+
+    Example:
+        .. code-block:: python
+
+            retriever = AmazonKendraRetriever(
+                index_id="c0806df7-e76b-4bce-9b5c-d5582f6b1a03"
+            )
+
+    """
+
+    index_id: str
```

Comment:
note BaseRetriever doesn't inherit from pydantic's BaseModel, so these are all just class attributes

Reply:
You are right, we can get rid of these. Let me update this and re-submit.